llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -974,11 +974,13 @@ void GCNSchedStage::checkScheduling() {
                       << DAG.MinOccupancy << ".\n");
   }
 
-  unsigned MaxVGPRs = ST.getMaxNumVGPRs(MF);
+  unsigned MaxVGPRs = ST.getMaxNumVGPRs(MF, /*WholeRegisterFile*/ true);
+  unsigned MaxArchVGPRs = ST.getMaxNumVGPRs(MF, /*WholeRegisterFile*/ false);
```

Comment:
Do I get this right, after the last change `MaxArchVGPRs` is initialized with the equivalent of `min(MaxVGPRs, getAddressableNumArchVGPRs)`? If so it is easier to do just this.

Reply:
This is correct, but I think other users of `getMaxNumVGPRs` may want to do the same thing

Reply:
The new interface is quite confusing and limited for just these few targets. Then I do not believe we have a lot of use for PSets since using our own RP tracker. If a small contained change can do the same thing, I would prefer a small contained change.

Reply:
Sounds good -- latest is the small contained change.